ciao-controller/openstack_volume.go

```diff
@@ -233,7 +233,36 @@ func (c *controller) ListVolumes(tenant string) ([]block.ListVolume, error) {
 }
 
 func (c *controller) ListVolumesDetail(tenant string) ([]block.VolumeDetail, error) {
-	return make([]block.VolumeDetail, 0), nil
+	var vols []block.VolumeDetail
+
+	devs, err := c.ds.GetBlockDevices(tenant)
+	if err != nil {
+		return vols, err
+	}
+
+	for _, data := range devs {
+		var vol block.VolumeDetail
+
+		vol.ID = data.ID
+		vol.Size = data.Size
+		vol.OSVolTenantAttr = data.TenantID
```

Comment:
The prior patch for ShowVolumeDetails() has a tenant check, ie:

    if data.TenantID != tenant {

Should this ListVolumesDetail() function also do so for symmetry and safety?

Or should neither function do the check, with both functions trusting that a function higher in the stack has validated that the passed tenant is reasonable and trusting that GetBlockDevices(tenant) will not return data not for the tenant?